aria/modeling/service.py

```diff
@@ -0,0 +1,1529 @@
+    :vartype outputs: {basestring: :class:`Parameter`}
+    :ivar operations: Custom operations that can be performed on the service
+    :vartype operations: {basestring: :class:`Operation`}
+    :ivar plugins: Plugins required to be installed
+    :vartype plugins: {basestring: :class:`Plugin`}
+    :ivar created_at: Creation timestamp
+    :vartype created_at: :class:`datetime.datetime`
+    :ivar updated_at: Update timestamp
+    :vartype updated_at: :class:`datetime.datetime`
+
+    :ivar permalink: ??
+    :vartype permalink: basestring
+    :ivar scaling_groups: ??
+    :vartype scaling_groups: {}
+
+    :ivar modifications: Modifications of this service
+    :vartype modifications: [:class:`ServiceModification`]
+    :ivar updates: Updates of this service
+    :vartype updates: [:class:`ServiceUpdate`]
+    :ivar executions: Executions on this service
+    :vartype executions: [:class:`Execution`]
+    """
+
+    __tablename__ = 'service'
+
+    @declared_attr
+    def service_template(cls):
+        return cls.many_to_one_relationship('service_template')
+
+    description = Column(Text)
+
+    @declared_attr
+    def meta_data(cls):
+        # Warning! We cannot use the attr name "metadata" because it's used by SqlAlchemy!
+        return cls.many_to_many_relationship('metadata', dict_key='name')
+
+    @declared_attr
+    def nodes(cls):
+        return cls.one_to_many_relationship('node')
+
+    @declared_attr
+    def groups(cls):
+        return cls.one_to_many_relationship('group')
+
+    @declared_attr
+    def policies(cls):
+        return cls.one_to_many_relationship('policy')
+
+    @declared_attr
+    def substitution(cls):
+        return cls.one_to_one_relationship('substitution')
+
+    @declared_attr
+    def inputs(cls):
+        return cls.many_to_many_relationship('parameter', table_prefix='inputs',
+                                             dict_key='name')
+
+    @declared_attr
+    def outputs(cls):
+        return cls.many_to_many_relationship('parameter', table_prefix='outputs',
+                                             dict_key='name')
+
+    @declared_attr
+    def operations(cls):
+        return cls.one_to_many_relationship('operation', dict_key='name')
+
+    @declared_attr
+    def plugins(cls):
+        return cls.many_to_many_relationship('plugin')
+
+    created_at = Column(DateTime, nullable=False, index=True)
+    updated_at = Column(DateTime)
+
+    # region orchestration
+
+    permalink = Column(Text)
+    scaling_groups = Column(modeling_types.Dict)
+
+    # endregion
+
+    # region foreign keys
+
+    __private_fields__ = ['substituion_fk',
+                          'service_template_fk']
+
+    # Service one-to-one to Substitution
+    @declared_attr
+    def substitution_fk(cls):
+        return cls.foreign_key('substitution', nullable=True)
+
+    # Service many-to-one to ServiceTemplate
+    @declared_attr
+    def service_template_fk(cls):
```

Comment:
service_template.services should return a dictionary (names as the keys) and not a list - same goes for all models with a unique name.